cts/cts-cli.in

```diff
@@ -1153,6 +1153,10 @@ class DatesRegressionTest(RegressionTest):
                  "iso8601 -d '2009-01-31 00:00:00Z' -D P2M -E '2009-03-31 00:00:00Z'"),
             Test("2009-01-31 + 3 Months",
                  "iso8601 -d '2009-01-31 00:00:00Z' -D P3M -E '2009-04-30 00:00:00Z'"),
+            Test("Test correct immediate plus-sign at timezone parsing",
```

Comment:
It seems like we should also test an immediate minus-sign, as well as leading spaces since we have to support that for a while.

I suspect there's still a lot of syntax that we either don't support but should, that we do support but shouldn't, etc. For example, I think Pacemaker will consider a plus or minus sign followed by a minus sign as valid, and treat the string after the  first plus/minus as a negative number. And we accept garbage after a `'Z'`. Etc.

Reply:
I have no doubt our iso8601 parsing leaves much to be desired, but I don't think it's worth fixing now since we have to support whatever old crud currently just happens to work.  One day when we can use a glib that was released less than 10 years ago, I'd like to take a wrecking ball to this entire area and use this instead:

https://docs.gtk.org/glib/ctor.DateTime.new_from_iso8601.html

Reply:
Added additional tests for immediate minus sign and versions with leading spaces.